design/one-pager-generic-usage-type.md

```diff
@@ -0,0 +1,397 @@
+# Generic `Usage` Type for Deletion Ordering
+
+* Owner: Hasan Türken (@turkenh)
+* Reviewers: @bobh66, @negz
+* Status: Draft
+
+## Background
```

Comment:
The background section is courtesy of Nic, I borrowed it from his great write-up in the description of the [proposal issue](https://github.com/crossplane/crossplane/issues/3393). 
Thank you @negz!